src/profile-logic/sanitize.js

```diff
@@ -317,12 +350,29 @@ function sanitizeThreadPII(
         if (
           currentMarker &&
           currentMarker.innerWindowID &&
-          windowIdToBeSanitized.has(currentMarker.innerWindowID)
+          windowIdFromPrivateBrowsing.has(currentMarker.innerWindowID)
         ) {
           // Remove any marker that we know they come from private browsing sessions
           markersToDelete.add(i);
         }
       }
+
+      if (PIIToBeRemoved.shouldKeepTabID !== null) {
```

Comment:
The if cases are a bit complex, but this is due to the fact that we want to remove markers that have no `innerWindowID` information.